src/en/quantumscans/src/eu/kanade/tachiyomi/extension/en/quantumscans/QuantumScans.kt

```diff
@@ -1,20 +1,132 @@
 package eu.kanade.tachiyomi.extension.en.quantumscans
 
-import eu.kanade.tachiyomi.multisrc.heancms.HeanCms
+import eu.kanade.tachiyomi.multisrc.iken.GenreFilter
+import eu.kanade.tachiyomi.multisrc.iken.Iken
+import eu.kanade.tachiyomi.multisrc.iken.SelectFilter
+import eu.kanade.tachiyomi.network.GET
 import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.parseAs
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import rx.Observable
 import java.util.concurrent.TimeUnit
 
-class QuantumScans : HeanCms(
-    "Quantum Scans",
-    "https://quantumscans.org",
+// Moved from HeanCms to Iken
+class QuantumScans : Iken(
+    "Quantum Toon",
     "en",
+    "https://quantumtoon.com",
+    "https://vapi.quantumtoon.com",
 ) {
-    // Moved from Keyoapp to HeanCms
-    override val versionId = 3
+    override val versionId = 4
 
     override val client = super.client.newBuilder()
         .rateLimit(3, 1, TimeUnit.SECONDS)
         .build()
 
-    override val useNewChapterEndpoint = true
+    override fun popularMangaRequest(page: Int): Request {
+        val url = "$apiUrl/api/query".toHttpUrl().newBuilder().apply {
+            addQueryParameter("page", page.toString())
+            addQueryParameter("perPage", "18")
+            addQueryParameter("orderBy", "totalViews")
+        }.build()
+        return GET(url, headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        return searchMangaParse(response)
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "$apiUrl/api/query".toHttpUrl().newBuilder().apply { // 'query' instead of 'posts'
+            addQueryParameter("page", page.toString())
+            addQueryParameter("perPage", "18")
+            addQueryParameter("orderBy", "updatedAt")
+        }.build()
+        return GET(url, headers)
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
```

Comment:
The result seems to be easier to parse (grepping `{}` objects per line) when you add header `rsc: 1`.

The data can be inspected with https://rsc-parser.vercel.app/